cmd/service-controller/definition_monitor.go

```diff
@@ -258,6 +285,61 @@ func (m *DefinitionMonitor) getServiceDefinitionFromAnnotatedDeployment(deployme
 	}
 }
 
+func (m *DefinitionMonitor) getServiceDefinitionFromAnnotatedDeploymentConfig(deploymentConfig *appv1.DeploymentConfig) (types.ServiceInterface, bool) {
+	var svc types.ServiceInterface
+	if protocol, ok := deploymentConfig.ObjectMeta.Annotations[types.ProxyQualifier]; ok {
+		port := map[int]int{}
+		if port = deducePortFromDeploymentConfig(deploymentConfig); len(port) > 0 {
+			svc.Ports = []int{}
+			for p, _ := range port {
+				svc.Ports = append(svc.Ports, p)
+			}
+		} else if protocol == "http" {
+			svc.Ports = []int{80}
+		} else {
+			event.Recordf(DefinitionMonitorIgnored, "Ignoring annotated deploymentconfig %s; cannot deduce port", deploymentConfig.ObjectMeta.Name)
+			return svc, false
+		}
+		svc.Protocol = protocol
+		if address, ok := deploymentConfig.ObjectMeta.Annotations[types.AddressQualifier]; ok {
+			svc.Address = address
+		} else {
+			svc.Address = deploymentConfig.ObjectMeta.Name
+		}
+
+		selector := ""
+		if deploymentConfig.Spec.Selector != nil {
+			selector = utils.StringifySelector(deploymentConfig.Spec.Selector)
+		}
+		svc.Targets = []types.ServiceInterfaceTarget{
+			{
+				Name:     deploymentConfig.ObjectMeta.Name,
+				Selector: selector,
+			},
+		}
+		if len(port) > 0 {
+			svc.Targets[0].TargetPorts = port
+		}
+		if labels, ok := deploymentConfig.ObjectMeta.Annotations[types.ServiceLabels]; ok {
+			svc.Labels = utils.LabelToMap(labels)
+		}
+		svc.Origin = "annotation"
+
+		if policyRes := m.policy.ValidateExpose("deploymentconfig", deploymentConfig.Name); !policyRes.Allowed() {
+			event.Recordf(DefinitionMonitorIgnored, "Policy validation error: deployment/%s cannot be exposed", deploymentConfig.ObjectMeta.Name)
```

Comment:
Minor: s/deployment/deploymentconfig/ for clarity.

Reply:
Updated in ea12208f8fc1859254e726a0bed7009981b40628